server/cpho/views/upload_indicators.py

```diff
@@ -0,0 +1,437 @@
+    #         else:
+    #             dimension_deduced_bool = True
+    #             dimension_deduced = geography
+    #             dimension_col_name = checking_column
+
+    #     if not dimension_deduced_bool:
+    #         print(
+    #             "WARNING ON LINE (",
+    #             csv_line_number,
+    #             "): DIMENSION NOT FOUND ",
+    #             "\nPLEASE CHECK RECORD: ",
+    #             record,
+    #         )
+    #         return None
+    #     return {
+    #         "dimension_col_name": dimension_col_name,
+    #         "dimension_value": dimension_deduced,
+    #     }
+    category_mapper = {
+        "": "",
+        "FACTORS INFLUENCING HEALTH": "factors_influencing_health",
+        "GENERAL HEALTH STATUS": "general_health_status",
+        "HEALTH OUTCOMES": "health_outcomes",
+    }
+
+    subcategory_mapper = {
+        "": "",
+        "SOCIAL FACTORS": "social_factors",
+        "HEALTH STATUS": "health_status",
+        "COMMUNICABLE DISEASES": "communicable_diseases",
+        "SUBSTANCE USE": "substance_use",
+        "CHILDHOOD AND FAMILY FACTORS": "childhood_and_family_risk_and_protective_factors",
+        "CHILDHOOD AND FAMILY RISK FACTORS": "childhood_and_family_risk_and_protective_factors",
+        "CHRONIC DISEASES AND MENTAL HEALTH": "chronic_diseases_and_mental_health",
+    }
+
+    data_quality_mapper = {
+        "": "",
+        "CAUTION": "caution",
+        "GOOD": "good",
+        "ACCEPTABLE": "acceptable",
+        "SUPPRESSED": "suppressed",
+        "VERY GOOD": "excellent",
+    }
+    value_unit_mapper = {
+        "": "",
+        "%": "%",
+        "PER 100,000": "per_100k",
+        "YEARS": "years",
+        "PER 1,000 CENSUS INHABITANTS": "per_100k_census",
+        "PER 10,000 PATIENT DAYS": "per_100k_patient_days",
+        "PER 100,000 LIVE BIRTHS": "per_100k_live_births",
+    }
+    dimension_type_mapper = {
+        "Province": DimensionType.objects.get(code="province"),
+        "Age Group": DimensionType.objects.get(code="age"),
+        "Sex": DimensionType.objects.get(code="sex"),
+        "Canada": DimensionType.objects.get(code="canada"),
+        "Region": DimensionType.objects.get(code="region"),
+        "Gender": DimensionType.objects.get(code="gender"),
+    }
+    # using combination of dimension type and value because provinces are regions as well; for now.
+    non_literal_dimension_value_mapper = {
+        "Province_ON": DimensionValue.objects.get(
+            dimension_type__code="province", value="on"
+        ),
+        "Province_AB": DimensionValue.objects.get(
+            dimension_type__code="province", value="ab"
+        ),
+        "Province_SK": DimensionValue.objects.get(
+            dimension_type__code="province", value="sk"
+        ),
+        "Province_MB": DimensionValue.objects.get(
+            dimension_type__code="province", value="mb"
+        ),
+        "Province_BC": DimensionValue.objects.get(
```

Comment:
Top level queries fail. For instance I can't even migrate or run any manage.py commands because these queries might fail. It's also an N+1 thing. Can we just refer to the codes, or use Q objects, e.g. `Q(dimension_type__code="province", value="sk")`

Reply:
Can these also be in a separate dict? They're kind of annoying to deal with inside the class, it's a lot of properties to collapse when trying to work with the class :)